File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {

        unordered_map<char , int>count;
        int l = 0;
        int maxf = 0;
        int res = 0;
        for(int r= 0 ; r < s.length() ; r++){

            count[s[r]]++;
            maxf = max(count[s[r]] , maxf);
            while((r-l+1) - maxf > k) {
                count[s[l]]--;
                l++;
            }

            res= max(maxf , r-l+1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isBalanced(root) {
        function checkHeight(root) {
            if (root === null) return 0;

            let lh = checkHeight(root.left);
            if (lh === -1) return -1;
            let rh = checkHeight(root.right);
            if (rh === -1) return -1;

            if (Math.abs(lh - rh) > 1) return -1;

            return 1 + Math.max(lh, rh);
        }

        const res = checkHeight(root);
        return res === -1 ? false : true;
    }
}
